src/ui/views/ViewServers.qml

```diff
@@ -88,9 +89,63 @@ Item {
                     }
                 }
 
+                VPNSearchBar {
+                    id: serverSearchInput
+
+                    anchors.left: parent.left
+                    anchors.right: parent.right
+                    anchors.leftMargin: Theme.vSpacing
+                    anchors.rightMargin: Theme.vSpacing
+                    enabled: true
+                    height: Theme.rowHeight
+                    onTextChanged: () => {
+                        countriesModel.invalidate();
+                    }
+                    placeholderText: VPNl18n.tr(VPNl18n.ServerViewSearchPlaceholder)
+                    stateError: countriesRepeater.count === 0
+
+                    RowLayout {
+                        id: searchWarning
+                        anchors.top: serverSearchInput.bottom
+                        anchors.topMargin: Theme.listSpacing
+                        visible: serverSearchInput.stateError
+
+                        VPNIcon {
+                            id: warningIcon
+
+                            source: "../resources/warning.svg"
+                            sourceSize.height: 14
+                            sourceSize.width: 14
+                        }
+
+                        VPNInterLabel {
+                            id: warningLabel
+                            color: Color.error.default
+                            text: VPNl18n.tr(VPNl18n.ServerViewSearchNoResultsLabel)
+                            font.pixelSize: Theme.fontSizeSmall
+                        }
+                    }
+                }
+
+                VPNFilterProxyModel {
+                    id: countriesModel
+                    source: VPNServerCountryModel
+                    filterCallback: country => {
+                        const searchString = serverSearchInput.text.toLowerCase();
+                        const includesSearchString = nameString => (
+                            nameString.toLowerCase().includes(searchString)
+                        );
+                        const includesName = includesSearchString(country.name);
+                        const includesLocalizedName = includesSearchString(country.localizedName);
+                        const matchesCountryCode = country.code.toLowerCase() === searchString;
+
+                        return includesName || includesLocalizedName || matchesCountryCode;
+                    }
+                }
+
                 Repeater {
-                    id: repeater
-                    model: VPNServerCountryModel
+                    id: countriesRepeater
```

Comment:
We'll need to update these lines to use this id too https://github.com/mozilla-mobile/mozilla-vpn-client/blob/7e4e981df5993cd0936da9ab2a137181df04ec03/src/ui/components/VPNServerCountry.qml#L41-L42